diff --git a/tests/python/tuples1_driver.py b/tests/python/tuples1_driver.py
index 4fb17d86..40af6f79 100644
--- a/tests/python/tuples1_driver.py
+++ b/tests/python/tuples1_driver.py
@@ -27,6 +27,9 @@ def ps_grammar(
     writes: List[Object], reads: List[Object], in_scope: List[Object], relaxed: bool
 ) -> Bool:
     ret_val = writes[0]
+    x, y = reads
+    x_tuple = make_tuple(x, x)
+    y_tuple = make_tuple(y, y)
     summary = choose(
         ret_val == tuple_mult(x_tuple) + tuple_mult(y_tuple),
         ret_val == tuple_mult(x_tuple) - tuple_mult(y_tuple),
